Sources/Purchasing/StoreKit1/ProductsFetcherSK1.swift

```diff
@@ -186,9 +186,16 @@ extension ProductsFetcherSK1: SKProductsRequestDelegate {
         }
     }
 
-    func clearCache() {
+    /// - Returns (via callback): The product identifiers that were removed, or empty if there were not
+    ///   cached products.
+    func clearCache(completion: ((Set<String>) -> Void)? = nil) {
         self.queue.async {
-            self.cachedProductsByIdentifier.removeAll()
+            let cachedProductIdentifiers = self.cachedProductsByIdentifier.keys
+            if !cachedProductIdentifiers.isEmpty {
+                Logger.debug(Strings.offering.product_cache_invalid_for_storefront_change)
+                self.cachedProductsByIdentifier.removeAll()
```

Comment:
Just to be sure, is there a possibility that there may not be the same number of products in different Storefronts?

Reply:
yep, well, `keepingCapacity: false` was already in SK2 and `false` is the default value, but I think it might be a good idea to explicitly state that we are completely clearing the storage 👍🏼

Reply:
Done ✅ ([0f59e57](https://github.com/RevenueCat/purchases-ios/pull/1743/commits/0f59e57cd4d305e2ebf3fc967dc33d0825c3c55c))